tasks/upload_processor.py

```diff
@@ -142,21 +175,26 @@ def process_impl_within_lock(
         commit_yaml,
         arguments_list,
         report_code,
+        parallel_idx=None,
+        in_parallel=False,
         **kwargs,
     ):
-        commit_yaml = UserYaml(commit_yaml)
-        log.info(
-            "Obtained upload processing lock, starting",
-            extra=dict(
-                repoid=repoid,
-                commit=commitid,
-                parent_task=self.request.parent_id,
-                report_code=report_code,
-            ),
-        )
+        if not PARALLEL_UPLOAD_PROCESSING_BY_REPO.check_value(repoid) and in_parallel:
```

Comment:
so if we AREN'T in the experiment but we WERE told to run this task in parallel? when would that happen? do we need this second condition?

Reply:
thanks for the catch, we don't need to second condition

Reply:
ah forgot the parenthesis: it should be `not check_value() or not in_parallel` which is equivalent to `not (check_value() and in_parallel)`

Reply:
the only time we don't take the lock is when `PARALLEL_UPLOAD_PROCESSING_BY_REPO.check_value(repoid) and in_parallel` is true